includes/classes/category_tree.php

```diff
@@ -137,64 +137,60 @@ function zen_category_tree($product_type = "all") {
-
-    for ($i=0; $i<$this->tree[$counter]['level']; $i++) {
-      if ($this->tree[$counter]['parent'] != (int)TOPMOST_CATEGORY_PARENT_ID) {
-        $this->categories_string .= CATEGORIES_SUBCATEGORIES_INDENT;
-      }
-    }
-
-
-    if ($this->tree[$counter]['parent'] == (int)TOPMOST_CATEGORY_PARENT_ID) {
-      $cPath_new = 'cPath=' . $counter;
-      $this->box_categories_array[$ii]['top'] = 'true';
-    } else {
-      $this->box_categories_array[$ii]['top'] = 'false';
-      $cPath_new = 'cPath=' . $this->tree[$counter]['path'];
-      $this->categories_string .= CATEGORIES_SEPARATOR_SUBS;
-    }
-    $this->box_categories_array[$ii]['path'] = $cPath_new;
-
-    if (isset($cPath_array) && in_array($counter, $cPath_array)) {
-      $this->box_categories_array[$ii]['current'] = true;
-    } else {
-      $this->box_categories_array[$ii]['current'] = false;
-    }
-
-    // display category name
-    $this->box_categories_array[$ii]['name'] = $this->categories_string . $this->tree[$counter]['name'];
+        $this->box_categories_array[$row]['current'] = 
+            (isset($cPath_array) && in_array($catID, $cPath_array)); 
 
-    // make category image available in case needed
-    $this->box_categories_array[$ii]['image'] = $this->tree[$counter]['image'];
+        // display category name
+        if ($this->tree[$catID]['parent'] == (int)TOPMOST_CATEGORY_PARENT_ID) { 
+           $this->box_categories_array[$row]['name'] = $this->tree[$catID]['name']; 
+        } else { 
+           $this->box_categories_array[$row]['name'] = str_repeat(CATEGORIES_SUBCATEGORIES_INDENT, (int)$this->tree[$catID]['level']) . CATEGORIES_SEPARATOR_SUBS . $this->tree[$catID]['name'];
+        }
 
-    if (zen_has_category_subcategories($counter)) {
-      $this->box_categories_array[$ii]['has_sub_cat'] = true;
-    } else {
-      $this->box_categories_array[$ii]['has_sub_cat'] = false;
-    }
+        // make category image available in case needed
+        $this->box_categories_array[$row]['image'] = 
+            $this->tree[$catID]['image'];
 
-    if (SHOW_COUNTS == 'true') {
-      $products_in_category = zen_count_products_in_category($counter);
-      if ($products_in_category > 0) {
-        $this->box_categories_array[$ii]['count'] = $products_in_category;
-      } else {
-        $this->box_categories_array[$ii]['count'] = 0;
-      }
-    }
+        $this->box_categories_array[$row]['has_sub_cat'] = 
+            zen_has_category_subcategories($catID); 
 
-    if ($this->tree[$counter]['next_id'] != false) {
-      $ii++;
-      $this->zen_show_category($this->tree[$counter]['next_id'], $ii);
+        if (SHOW_COUNTS == 'true') {
+            $products_in_category = zen_count_products_in_category($catID);
+            if ($products_in_category > 0) {
+                $this->box_categories_array[$row]['count'] = $products_in_category;
+            } else {
+                $this->box_categories_array[$row]['count'] = 0;
+            }

```

Comment:
Avoiding the unnecessary if/else

Will `zen_count_products_in_category()` ever give a value less than 0 ?   Is the check for 0 even needed?